### PR TITLE
[2.0.x] Improved SW micro stepping configuration

### PR DESCRIPTION
**Pull request by TheLongAndOnly**
Current micro stepping configuration is only limited to 2 pins. In addition there is only a fixed definition for the modes in Conditional_posts.h, which is not very flexible.

This pull requests extends implementation to 3 micro stepping pins (which many drivers require) and allows to define custom micro stepping pin settings in Configuration_adv.h.

### CODE REVIEW

File: Marlin/Configuration_adv.h [hunk 1]
```diff
@@ -458,6 +458,23 @@
...
 
+/**
+ * Custom Microstepping PIN settings
+ * Allows additional flexibility to defined the proper PIN settings for different 
+ * stepper drivers. Defaults are normally applied in Conditional_post.h
+ * Uncomment MICROSTEP_CUSTOM to enable manual definition and ensure all microstep modes selected below
+ * are defined. At max, 3 MS PINS are supported, if less are defined, the last settings are ignoreds
+ */
+//#define MICROSTEP_CUSTOM
+#ifdef MICROSTEP_CUSTOM
```
Comment (thinkyhead) [inline Marlin/Configuration_adv.h]: Required for Marlin:
```cpp
#if ENABLED(MICROSTEP_CUSTOM)
```

File: Marlin/src/inc/Conditionals_post.h [hunk 2]
```diff
@@ -222,24 +222,26 @@
...
-  #define MICROSTEP2 HIGH,LOW
-  #define MICROSTEP4 LOW,HIGH
-#endif
-#define MICROSTEP8 HIGH,HIGH
-#ifdef __SAM3X8E__
-  #if MB(ALLIGATOR)
-    #define MICROSTEP16 LOW,LOW
-    #define MICROSTEP32 HIGH,HIGH
+// MS1 MS2 MS3 Stepper Driver Microstepping mode table
+#ifndef MICROSTEP_CUSTOM
```
Comment (thinkyhead) [inline Marlin/src/inc/Conditionals_post.h]:
```cpp
#if DISABLED(MICROSTEP_CUSTOM)
```

**Comment by TheLongAndOnly** [comment]
Thanks for the feedback. I changed this and added a few more checks to ensure backwards compatibility in case only 2 micro stepping pins are defined.

**Comment by TheLongAndOnly** [comment]
Same here as other pull requests currently failing: `error in void CardReader::removeJobRecoveryFile() "success" not defined.`

**Comment by Allted** [comment]
This looks like it could fix the last of the major issues with getting the Archim1 boards ready for Marlin 2.0. Would it help if I updated all the configs or is there a script for that sort of thing?

**Comment by TheLongAndOnly** [comment]
Strangly I cannot see this comment in Github. So replying to the Mail.

This is pretty old code and I would need to rebase it as many things have changed, which partially already cover this PR. Give me a moment to check.

**Comment by Allted** [comment]
Sorry, I deleted the comment as I realized it already seems to be in the bugfix branch. I guess it got pulled between my updates. I think this can be closed now?

**Pull request closed without merging**